python/hail/dataset.py

```diff
@@ -915,6 +915,256 @@ def annotate_variants_vds(self, other, expr=None, root=None):
+
+        # load database with annotation metadata
+        with hadoop_read('gs://annotationdb/annotationdb.sql') as f:
+            curs = conn.executescript(f.read())
+
+        # parameter substitution string to put in SQL query
+        like = ' OR '.join('a.annotation LIKE ?' for x in annotations)
+
+        # query to extract path of all needed database files and their respective annotation exprs 
+        qry = """SELECT file_path, annotation, file_type, file_element, f.file_id
+                 FROM files AS f INNER JOIN annotations AS a ON f.file_id = a.file_id
+                 WHERE {}""".format(like)
+
+        # run query and collect results in a file_path: expr dictionary
+        results = curs.execute(qry, [x + '%' for x in annotations]).fetchall()
+
+        # all file_ids to be used
+        file_ids = tuple(set([x[4] for x in results]))
+
+        # parameter substitution string
+        sub = ','.join('?' for x in file_ids)
+
+        # query to fetch count of total annotations in each file
+        qry = """SELECT file_path, COUNT(*)
+                 FROM files AS f INNER JOIN annotations AS a ON f.file_id = a.file_id
+                 WHERE f.file_id IN ({})
+                 GROUP BY file_path""".format(sub)
+            
+        # collect counts in file_id: count dictionary
+        cnts = {x[0]: x[1] for x in curs.execute(qry, file_ids).fetchall()}
+
+        # collect dictionary of file_path: expr entries
+        file_exprs = {}
+        for r in results:
+            expr = r[1] + '=' + 'table' if r[2] == 'table' and cnts[r[0]] < 2 else r[1] + '=' + r[2] + '.' + r[3]
+            try:
+                file_exprs[r[0]] += ',' + expr
+            except KeyError:
+                file_exprs[r[0]] = expr
+
+        # are there any gene annotations?
+        are_genes = any([x.startswith('gs://annotationdb/gene/') for x in file_exprs])
+
+        # subset to VEP annotations
+        veps = any([x == 'vep' for x in file_exprs])
+
+        # if VEP annotations are selected, or if gene-level annotations are selected with no specified gene_key, annotate with VEP
+        if veps or (are_genes and not gene_key):
+
+            # VEP annotate the VDS
+            self = self.vep(config = '/vep/vep-gcloud.properties', root = 'va.vep')
+
+            # extract 1 gene symbol per variant from VEP annotations if a gene_key parameter isn't provided
+            if are_genes:
+
+                self = (
+                    self
+                    .annotate_variants_expr(
+                        """
+                        va.gene.consequence = 
+                            let csq_terms = va.vep.transcript_consequences.filter(t => t.canonical == 1).map(t => t.consequence_terms.toSet) in
+                            if (isMissing(csq_terms))
+                                va.vep.most_severe_consequence
+                            else if (csq_terms.exists(t => t.contains("transcript_ablation")))
```

Comment:
we can make this code a lot shorter.

First, annotate global:
```python
csqs = ['transcript_ablation', 'splice_acceptor_variant', ...]
self = self.annotate_global('global.csq_rank',  csqs, TArray(TString()))
```
Now use this global annotation and `find` to avoid a bunch of if/else
```
let csq_terms = va.vep.transcript_consequences
    .filter(t => t.canonical == 1)
    .flatMap(t => t.consequence_terms).toSet() in 
if (isMissing(csq_terms)
    va.vep.most_severe_consequence
else 
    orElse(
        global.csq_rank.find(term => csq_terms.contains(term)),
        va.vep.most_severe_consequence) 
```